javascript-bank/bank.js

```diff
@@ -0,0 +1,35 @@
+/* exported Bank */
+function Bank() {
+  this.nextAccountNumber = 1;
+  this.accounts = [];
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  if (balance <= 0 || !Number.isInteger(balance)) {
+    return null;
+  }
+  var openAccount = new Account(this.nextAccountNumber, holder);
+
+  openAccount.transactions.push({ type: 'deposit', amount: balance });
```

Comment:
This shouldn't be done like this, you should use the `deposit` method on the `openAccount` object.